src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp

```diff
@@ -387,14 +443,23 @@ bool SmokeCoAlarmServer::HandleRemoteSelfTestRequest(CommandHandler * commandObj
         }
         else
         {
-            success = opHandler(endpointId);
-            if (!success)
+            success = SetExpressedState(endpointId, ExpressedStateEnum::kTesting);
+            if (success)
             {
-                status = Status::Failure;
+                success = SetTestInProgress(endpointId, true);
+            }
+
+            if (success)
+            {
+                success = emberAfPluginSmokeCoAlarmSelfTestRequestCommand(endpointId);
+
+                UnsetExpressedState(endpointId, ExpressedStateEnum::kTesting);
+                SetTestInProgress(endpointId, false);
```

Comment:
Again, the app needs to do that.  We don't know when the test is done.

If we want to make this easier, we could have a function for "test is done, here is the new expressed state" that the app would call, but all that would do is SetExpressedState and SetTestInProgress(false).

Reply:
OK, change to APP to set the state.